docs: Remove trailing newline from the code of Playground links

fasttime opened the pull request:
<!--
    Thank you for contributing!

    ESLint adheres to the [JS Foundation Code of Conduct](https://eslint.org/conduct).
-->

#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

<!--
    The following template is intentionally not a markdown checkbox list for the reasons
    explained in https://github.com/eslint/eslint/pull/12848#issuecomment-580302888
-->

[X] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

<!--
    If the item you've checked above has a template, please paste the template questions below and answer them. (If this pull request is addressing an issue, you can just paste a link to the issue here instead.)
-->

Fixes #17627

<!--
    Please ensure your pull request is ready:

    - Read the pull request guide (https://eslint.org/docs/latest/contribute/pull-requests)
    - Include tests for this change
    - Update documentation for this change (if appropriate)
-->

<!--
    The following is required for all pull requests:
-->

#### What changes did you make? (Give an overview)

Removed the last newline character from the code of Playground links in rule examples.

Fixed examples of the rules `eol-last` and `no-multiple-empty-lines`.

#### Is there anything you'd like reviewers to focus on?

<!-- markdownlint-disable-file MD004 -->

fasttime commented on docs/src/rules/no-multiple-empty-lines.md:
````diff
@@ -88,34 +88,17 @@ var bar = 3;
-
-```
-
-:::
````
After removing the `⏎` characters that are causing syntax errors, this code block becomes identical to [the previous incorrect example block](https://github.com/eslint/eslint/pull/17641/files#diff-bb702378ac043bca3c41565355c85509c8bf26510156a27be567f42eb10744cdR59-R72).

mdjermanovic replied:
I think we should keep examples with `⏎`. We could just remove `::: correct`/`::: incorrect`.

mdjermanovic replied:
And now that there's no extra `\n`, we could add another `::: correct` example after the existing one (the one that ends on line 87), just with another `\n` so that it looks like the existing example + line 7 that appears as an empty line.

fasttime replied:
If we are going to [replace `<<TAB>>` with a tab character](https://github.com/eslint/eslint/issues/17629#issuecomment-1754858588) in Playground links, maybe we could also replace `⏎` with an empty string (since there is already a newline), or use a more similar placeholder like `<<NEWLINE>>` or `<<LINEBREAK>>` to visually denote a newline where it makes sense for clarity, and remove those sequences in the code that gets passed to the Playground.

Then there could be just one code block that presents nicely in rule docs pages and has a functioning "Open in Playground" button. The intent is to avoid duplications (both in source code and in reading material, I guess).

mdjermanovic replied:
Replacing `⏎` with an empty string sounds good to me.

Tanujkanti4441 replied:
I think we should keep at least one example code with `⏎` just for the understanding sake without making it a playground link or without  `::: correct`/::: `incorrect`

fasttime replied:
Hi @Tanujkanti4441, thanks for the feedback. I think the assumption here is that the `⏎` characters would be removed only from the code that gets passed to the Playground. They would remain visible in the code blocks on the documentation pages along with the correct/incorrect badge and the "Open in Playground" button. Does this sound like a viable solution to you?

Tanujkanti4441 replied:
Yes! This sounds great

fasttime commented:
I've updated the Playground link logic and added `⏎` to all code examples in `no-multiple-empty-lines` with option `maxEOF`. Please check if this is looking good now: https://deploy-preview-17641--docs-eslint.netlify.app/rules/no-multiple-empty-lines.
